src/app/application/components/app-collection/AppCollection.vue

```diff
@@ -13,34 +13,39 @@
       })
       return {data: items}
     }"
-    :cell-attrs="({ headerKey }: any) => {
-      return {
-        class: {
-          [`${headerKey}-column`]: true,
-        },
-      }
-    }"
+    :cell-attrs="getCellAttributes"
     empty-state-icon-size="96"
     disable-sorting
     hide-pagination-when-optional
     @row:click="click"
   >
     <template
-      v-for="key in Object.keys(slots)"
-      #[`${key}`]="{ row }"
+      v-if="slots.toolbar"
+      #toolbar
+    >
+      <div class="app-collection-toolbar">
+        <slot name="toolbar" />
+      </div>
+    </template>
+
+    <template
+      v-for="key in Object.keys(remainingSlots)"
+      #[`${key}`]="{ row, rowValue }"
```

Comment:
Could we just check if `key` equals `toolbar` and if so add the wrapping div? In fact from what I can see the KTable toolbar already has a wrapping `k-table-toolbar`, do we need to add yet another one?

Reply:
> Could we just check if key equals toolbar and if so add the wrapping div?

Technically, yes, but I don’t see how that’s better. Leaving this as is.

> In fact from what I can see the KTable toolbar already has a wrapping k-table-toolbar, do we need to add yet another one?

`.k-table-toolbar` has styles attached to it that I don’t want to re-use. Specifically that every element in toolbar should have `display: flex` is not desirable in my book. Having a separate element here makes the layout of toolbar items robust and independent from changes in Kongponents.

Reply:
> .k-table-toolbar has styles attached to it that I don’t want to re-use. Specifically that every element in toolbar should have display: flex is not desirable in my book.

I'm not seeing this could you post a screengrab or something showing me these?

![Screenshot 2023-07-19 at 08 57 11](https://github.com/kumahq/kuma-gui/assets/554604/edff5b01-7eae-4e0c-982e-15827feb7ba9)

lmk

Reply:
You have access to https://github.com/Kong/kongponents, right?

https://github.com/Kong/kongponents/blob/main/src/components/KTable/KTable.vue#L986-L992

Reply:
Yes I do, but thanks for linking thats super helpful